src/pages/basket/listeners/applyPromoCode.ts

```diff
@@ -0,0 +1,45 @@
+import { CartGetManager } from '../../../server/cart/getCartById';
+import { isPromoLap } from '../functions/isPromoLap';
+import { recalculatePromo } from './changeQuantity/recalculatePromo';
+import { recalculateValues } from './changeQuantity/recalculateValues';
+
+export function applyPromoCode() {
+    document.addEventListener('click', async (event) => {
+        const target = event.target as HTMLButtonElement;
+        if (target.id.includes('-lap-but')) {
+            const id = target.id.replace('-lap-but', '');
+            const input = document.getElementById(`${id}-promo-input`) as HTMLInputElement;
+            const price = document.getElementById(`${id}-cart-prod-price`) as HTMLDivElement;
+            const prodName = document.getElementById(`${id}-cart-prod-name`) as HTMLDivElement;
+            // const totalSumBlock = document.getElementById('cart-sum') as HTMLDivElement;
```

Comment:
no commented code
split code into some functions, it is very long

Reply:
corrected